TypeScript errors in explore/content.tsx (ReactNode type mismatches)

## TypeScript Errors in explore/content.tsx

### Problem
Multiple TypeScript type errors exist in `apps/web/app/explore/content.tsx` that are currently masked by `ignoreBuildErrors` setting.

### Affected File & Errors

**File:** `apps/web/app/explore/content.tsx`

#### Error 1 (Line 379)
```
error TS2322: Type 'unknown' is not assignable to type 'ReactNode'.
```

#### Error 2 (Line 392)
```
error TS2322: Type 'unknown' is not assignable to type 'ReactNode'.
```

#### Error 3 (Line 1480)
```
error TS2322: Type '(question: string) => void' is not assignable to type 'MouseEventHandler<HTMLButtonElement>'.
Types of parameters 'question' and 'event' are incompatible.
Type 'MouseEvent<HTMLButtonElement, MouseEvent>' is not assignable to type 'string'.
```

### Impact
- Type safety is compromised in Explore page
- Potential runtime errors when rendering unknown types as ReactNode
- Event handler type mismatch may cause unexpected behavior

### Suggested Fix

1. **For lines 379 and 392 (unknown to ReactNode)**: 
   - Add proper type guards before rendering:
   ```tsx
   {typeof value === 'string' || typeof value === 'number' ? value : null}
   ```
   - Or use type assertion if the type is known:
   ```tsx
   {value as ReactNode}
   ```
   - Better: Fix the source to return proper types instead of `unknown`

2. **For line 1480 (MouseEventHandler mismatch)**:
   - The handler signature is wrong. Should be:
   ```tsx
   // Wrong:
   const handleClick = (question: string) => { ... }
   
   // Correct:
   const handleClick = (event: MouseEvent<HTMLButtonElement>) => { ... }
   // Or if you need the question:
   const handleClick = (question: string) => (event: MouseEvent<HTMLButtonElement>) => { ... }
   ```
   - Or use `onClick={() => handleQuestion('some question')}` pattern

### Related
- #2287 (Remove console.log statement in production code - apps/web/app/explore/content.tsx)
- #2294 (Critical: ignoreBuildErrors masks TypeScript errors in production builds)

--- a/apps/web/app/explore/content.tsx
+++ b/apps/web/app/explore/content.tsx
@@ -376,7 +376,7 @@ function renderExplorerCell(column: string, value: unknown) {
         href={`/analyze/${value}`}
         className="text-[#fbe593] transition-colors hover:text-[#fceeb4]"
       >
-        {value}
+        {String(value)}
       </Link>
     );
   }
@@ -389,7 +389,7 @@ function renderExplorerCell(column: string, value: unknown) {
         target="_blank"
         rel="noopener noreferrer"
       >
-        {value}
+        {String(value)}
       </a>
     );
   }
@@ -1480,7 +1480,7 @@ function ExploreExecutionLayout({
                 <div className="mt-3 flex items-center gap-3">
                   <button
                     type="button"
-                    onClick={onSubmit}
+                    onClick={() => onSubmit(input)}
                     className="rounded-lg bg-[#6a2c2f] px-3 py-1.5 text-xs font-medium text-white hover:bg-[#7d3437] transition-colors"
                   >
                     Try again

